service/history/workflowTaskHandlerCallbacks.go

```diff
@@ -383,6 +384,7 @@ func (handler *workflowTaskHandlerCallbacksImpl) handleWorkflowTaskCompleted(
 	if err != nil {
 		return nil, err
 	}
+	defer workflowContext.GetContext().UpdateRegistry().Prune()
```

Comment:
I believe it is not correct to call `Prune` here. It is race condition. What if handler code completes before API caller read update result?

Reply:
By the time this runs, I _believe_ (but may be wrong!) that (1) updates completed in this WFT will be written to MS as EventPointers, and (2) updates completed in this WFT will deliver an updatepb.Completed message to registered update.Update objects in the update.Registry. If those two things are correct then a Poll call through the API either registers its interest _before_ Prune() in which case it will block on the Future in update.Update, or _after_ Prune() in which case it will "miss" the update.Update object but will then find through MutableState/History.

Have I missed something?